### PR TITLE
Deprecate invoking Rally without a subcommand

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we deprecate running Rally without a subcommand. This
makes the user's intention explicit and avoids any leniency in Rally
around subcommand handling. Once a subcommand is required, command line
help can also be improved by only showing command line flags relevant to
a specific subcommand.

Relates #1142

### CODE REVIEW

File: esrally/rally.py
Comment (dliappis) [inline esrally/rally.py:592]: I wonder, have we ever considered https://docs.python.org/3/library/warnings.html for such warnings?

Comment (dliappis) [reply esrally/rally.py:592]: Should we specify that the subcommand `race` will be required starting with Rally 2.1.0?

**Review by dliappis — APPROVED** [review]
LGTM left a question and a suggestion.

File: esrally/rally.py
Comment (danielmitterdorfer) [reply esrally/rally.py]: I attemped to use the warnings module now but there are two issues:

1. We configure Rally to redirect warnings to the log. We do this because of urllib3 which would otherwise write to the terminal (see https://github.com/elastic/rally/blob/5bf0fed3ddf442d19265d314f8d50bce95ad4127/esrally/log.py#L129).
2. Even if we disable this behavior, the output is slightly verbose:

```
/Users/daniel/Projects/rally/esrally/rally.py:592: FutureWarning: Invoking Rally without a subcommand is deprecated. Specify the 'race' subcommand explicitly.
  warnings.warn("Invoking Rally without a subcommand is deprecated. Specify the 'race' subcommand explicitly.", category=FutureWarning)
```

I've included a note about 2.1.0 in the message in https://github.com/elastic/rally/pull/1155/commits/7513e561a0734a516801d17858b4e400b15e37bf.

Comment (dliappis) [reply esrally/rally.py]: Thanks for taking a look at the warnings module! This was just something to consider, given the ^^ observations doesn't make sense to use in this project right now.

Also thanks for enhancing the message in https://github.com/elastic/rally/commit/7513e561a0734a516801d17858b4e400b15e37bf, I think it'll help the user base prepare promptly before upgrading to the next minor.

Comment (danielmitterdorfer) [reply esrally/rally.py]: Well, thanks for bringing it up! I'll wait until the CI build is finished and will merge the PR then.

**Pull request merged**
